### ISSUE TITLE
SSH CA: rsa-sha1 should not be default signing algorithm  

### ISSUE THREAD

**Issue by gargath** [hashicorp/vault#11608]
**Describe the bug**
Modern-ish versions of sshd refuse SSH CA certificates signed with sha1. https://github.com/hashicorp/vault/pull/9096 added support for providing a `algorithm_signer` attribute to override this, but sha1 is still the default if this is not specified.

This means that following the documentation at https://www.vaultproject.io/docs/secrets/ssh/signed-ssh-certificates#signing-key-role-configuration will result in certificates that will be rejected by most hosts.

The documentation should outline the use of the `algorithm_signer` attribute and include it in the sample role definitions provided.
In addition, maybe `sha1` should no longer be the default?

**To Reproduce**
1. With a fresh, unsealed vault, follow documentation at https://www.vaultproject.io/docs/secrets/ssh/signed-ssh-certificates#signing-key-role-configuration
2. Note that Step 4 in the guide does not mention the `algorithm_signer`
3. Configure a recent ubuntu machine to trust the SSH CA
4. Sign a key with ault write ssh-client-signer/sign/<some_fole> public_key=@$HOME/.ssh/id_rsa.pub
5. Try to log into the ubuntu machine with the signed key and cert
6. Observe sshd logs to see error: `sshd[60694]: userauth_pubkey: certificate signature algorithm ssh-rsa: signature algorithm not supported [preauth]`

**Expected behavior**
The certificate should be signed with an algorithm that is supported by modern sshd implementations, e.g. `rsa-sha2-512`.

**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.7.0
* Vault CLI Version (retrieve with `vault version`):  v1.7.1 ('917142287996a005cb1ed9d96d00d06a0590e44e+CHANGES')
* Server Operating System/Architecture: client: MacOS Cataline (10.15.7), server: Ubuntu 20.04.2 LTS

**Additional context**
Using this role instead of the one in the documentation worked as expected:
```
$ vault write ssh-client-signer/roles/my-role -<<"EOH"
{
  "allow_user_certificates": true,
  "allowed_users": "*",
  "allowed_extensions": "permit-pty,permit-port-forwarding",
  "default_extensions": [
    {
      "permit-pty": ""
    }
  ],
  "key_type": "ca",
  "default_user": "ubuntu",
  "algorithm_signer": "rsa-sha2-512",
  "ttl": "30m0s"
}
EOH
```

**Comment by marcpaulchand** [comment]
I agree.
Here are the release note for openSSH 8.2 [1]
```
It is now possible[1] to perform chosen-prefix attacks against the
SHA-1 hash algorithm for less than USD$50K. For this reason, we will
be disabling the "ssh-rsa" public key signature algorithm that depends
on SHA-1 by default in a near-future release.
```
Until changes, it's still possible to bring back the deprecated algo with :

```
CASignatureAlgorithms ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-ed25519,rsa-sha2-512,rsa-sha2-256,ssh-rsa
```
in your `sshd_config`
[1]: https://www.openssh.com/txt/release-8.2

**Comment by marcpaulchand** [comment]
My bad, in fact everything is fine.
Here is vault documentation [1]

There is an `algorithm_signer` role option.
So with the option it gives : 
```
ault write ssh-client-signer/roles/test -<<"EOH"
{
  "allow_user_certificates": true,
  "allowed_users": "*",
  "allowed_extensions": "permit-pty,permit-port-forwarding",
  "default_extensions": [
    {
      "permit-pty": ""
    }
  ],
  "key_type": "ca",
  "default_user": "test",
  "ttl": "30m0s",
  "algorithm_signer": "rsa-sha2-512"
}
EOH
```
and no more problem.

This should satisfy everyone.

[1] : https://www.vaultproject.io/api/secret/ssh#algorithm_signer

**Comment by digitalfredy** [comment]
I agree that Vault default conf must at least to be in line with recent OpenSsh default conf because I'm following the [Vault doc](https://www.vaultproject.io/docs/secrets/ssh/signed-ssh-certificates) and I waste a lot of time because it was not working and last thing in my list was a "insecure" configuration by default when following HC documentation.

**Comment by cipherboy** [comment]
\o Hello @gargath and others -- I've recently fixed this on `main` ahead of a future v1.10 release. If anyone has time to test it, would appreciate some more eyes before we cut releases. :-) This was done in conjunction with #11035 (fixed in #14008) so that there should be plenty of possibilities for broader OpenSSH version support.
